productivity-suite/app/fragments/todo-lists/src/root.tsx

```diff
@@ -31,30 +28,76 @@ export const Root = component$(() => {
     selectedListName: "",
   });
 
-  useMount$(async () => {
-    state.currentUser = envCurrentUser;
-    state.todoLists = initialUserData.todoLists;
-    if (initialSelectedListName) {
-      const idx = state.todoLists.findIndex(
-        ({ name }) => name === initialSelectedListName
-      );
-      state.idxOfSelectedList = idx !== -1 ? idx : 0;
-    }
+  const requestCookie: string = useEnvData("requestCookie")!;
+
+  useServerMount$(async () => {
+    const currentUser =
+      getBus().latestValue<{ username: string }>("authentication")?.username ??
+      null;
+
+    state.currentUser = currentUser ?? undefined;
+
+    const todoLists = currentUser
+      ? await getTodoLists(currentUser, requestCookie)
+      : null;
+    state.todoLists = todoLists ?? [];
+
+    const selectedListName =
+      getBus().latestValue<{ name: string }>("todo-list-selected")?.name ??
+      null;
+
+    const idx = state.todoLists.findIndex(
+      ({ name }) => name === selectedListName
+    );
+
+    state.idxOfSelectedList = idx !== -1 ? idx : state.todoLists.length - 1;
+    state.selectedListName = state.todoLists[idx]!.name;
   });
 
   const ref = useSignal<Element>();
 
+  useClientEffect$(
+    () => {
+      if (ref.value) {
+        getBus(ref.value).listen<{ name: string }>(
+          "todo-list-selected",
+          (listDetails) => {
+            const newIdxOfSelectedList = state.todoLists.findIndex(
+              ({ name }) => name === listDetails?.name
+            );
+            if (newIdxOfSelectedList >= 0) {
+              state.idxOfSelectedList = newIdxOfSelectedList;
+              state.selectedListName =
+                state.todoLists[newIdxOfSelectedList].name;
+            }
+          }
+        );
+      }
+    },
+    { eagerness: "load" }
```

Comment:
Maybe add a comment here explaining this workaround?

Reply:
yeah I thought about that too at some point (but I wasn't sure if it would add noise) 😕

I'll add it 🙂👍

Reply:
I've added a comment here and in the outlet component when we emit the `qinit`, I kept the comments short and added a link to the issue you've opened, please let me know if you think it's ok 🙂